stellargraph/core/graph.py

```diff
@@ -242,11 +242,25 @@ def _convert_from_node_data(data, node_type_map, node_types, dtype="f"):
     return data_index, data_arrays
 
 
-class StellarGraphBase:
+class StellarGraphFactory(type):
     """
-    StellarGraph class for undirected graph ML models. It stores both
-    graph information from a NetworkX Graph object as well as features
-    for machine learning.
+    Private class for instantiating the StellarGraph interface from
+    user-supplied information.
+    """
+
+    def __call__(cls, *args, **kwargs):
+        if cls is StellarGraph:
+            return NetworkXStellarGraph(*args, is_directed=False, **kwargs)
```

Comment:
I agree, I left it up to the Python infrastructure to complain, since `is_directed` was left undocumented. The other thing we could do is raise an error saying to not explicitly set `is_directed`.

Reply:
I think raising an error if `is_directed` and the class-specified directedness don't agree would be good.